src/plugins/default/DuplicateCheckerPlugin.ts

```diff
@@ -1,50 +1,70 @@
-  public priority: number = 40;
+  public priority: number = 90;
   public name: string = 'DuplicateCheckerPlugin';
 
+  private _mergedEvents: MergedEvent[] = [];
   private _processedHashcodes: TimestampedHash[] = [];
   private _getCurrentTime: () => number;
+  private _interval: number;
 
-  constructor(getCurrentTime: () => number = () => Date.now()) {
+  constructor(getCurrentTime: () => number = () => Date.now(), interval: number = 60000) {
     this._getCurrentTime = getCurrentTime;
+    this._interval = interval;
+
+    setInterval(() => {
+      while (this._mergedEvents.length > 0) {
+        this._mergedEvents.shift().resubmit();
+      }
+    }, interval);
   }
 
   public run(context: EventPluginContext, next?: () => void): void {
-    function isDuplicate(error: IInnerError, processedHashcodes, now, log: ILog): boolean {
-      while (error) {
-        let hashCode = Utils.getHashCode(error.stack_trace && JSON.stringify(error.stack_trace));
+    function getHashCode(error: IInnerError): number {
+      let hashCode = 0;
 
-        // Only process the unique errors times within a 2 second window.
-        if (hashCode && processedHashcodes.some(h => h.hash === hashCode && h.timestamp >= (now - 2000))) {
-          log.info(`Ignoring duplicate error event hash: ${hashCode}`);
-          return true;
+      while (error) {
+        if (error.stack_trace && error.stack_trace.length) {
+          hashCode = (hashCode * 397) ^ Utils.getHashCode(JSON.stringify(error.stack_trace));
         }
+        error = error.inner;
+      }
 
-        // Add this exception to our list of recent processed errors.
-        processedHashcodes.push({ hash: hashCode, timestamp: now });
+      return hashCode;
+    }
 
-        // Only keep the last 20 recent errors.
-        while (processedHashcodes.length > 20) {
-          processedHashcodes.shift();
-        }
+    let error = context.event.data['@error'];
+    let hashCode = getHashCode(error);
+    if (!hashCode) {
+      return;
+    }
 
-        error = error.inner;
-      }
+    let count = context.event.count || 1;
+    let now = this._getCurrentTime();
 
-      return false;
+    let merged = this._mergedEvents.filter(s => s.hashCode === hashCode)[0];
+    if (merged) {
+      merged.incrementCount(count);
+      merged.updateDate(context.event.date);
+      context.cancelled = true;
+      return;
     }
 
-    if (context.event.type === 'error') {
-      if (isDuplicate(context.event.data['@error'], this._processedHashcodes, this._getCurrentTime(), context.log)) {
-        context.cancelled = true;
-        return;
-      }
+    if (this._processedHashcodes.some(h => h.hash === hashCode && h.timestamp >= (now - this._interval))) {
+      this._mergedEvents.push(new MergedEvent(hashCode, context, count));
+      context.cancelled = true;
+      return;
+    }
+
+    this._processedHashcodes.push({ hash: hashCode, timestamp: now });
```

Comment:
would be nice if we added some minimal logging to this class. Makes it much easier to debug issues